### PR TITLE
Keep source reserved memory after vm clone

### PR DESCRIPTION
**Pull request by futuretea**
**IMPORTANT: Please do not create a Pull Request without creating an issue first.**

Keep source reserved memory after vm clone

**Related Issue:**
https://github.com/harvester/harvester/issues/3912

**Test plan:**

1. Go to create a vm
1. Provide a reserved memory count for the vm (good for windows vms)
1. Clone the vm (clone volume data mode)
1. Check both `edit as yaml` and also just the `config -> advanced options -> show more`

### CODE REVIEW

File: pkg/api/vm/handler.go [hunk 1]
```diff
@@ -1244,6 +1244,9 @@ func getClonedVMYamlFromSourceVM(newVMName string, sourceVM *kubevirtv1.VirtualM
 		},
 		Spec: *sourceVM.Spec.DeepCopy(),
 	}
+	if sourceReservedMemory := sourceVM.Annotations[util.AnnotationReservedMemory]; sourceReservedMemory != "" {
+		newVM.Annotations[util.AnnotationReservedMemory] = sourceReservedMemory
+	}
```
Comment (guangbochen) [inline pkg/api/vm/handler.go]: let's define a constant array key and loop to add if they exist for better extendability, thanks.

Comment (futuretea) [reply pkg/api/vm/handler.go]: updated, add a slice var ([]string cannot be a const)

**Review by guangbochen — APPROVED** [review]
LGTM, thanks.

**Review by bk201 — APPROVED** [review]

**Comment by bk201** [comment]
I guess we also need to backport.

**Comment by futuretea** [comment]
@mergify backport v1.1

**Pull request merged**